lib/compat/wordpress-6.8/class-gutenberg-rest-post-counts-controller.php

```diff
@@ -0,0 +1,217 @@
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$post_type = get_post_type_object( $request['post_type'] );
+
+		if ( ! $post_type ) {
+			return new WP_Error(
+				'rest_invalid_post_type',
+				__( 'Invalid post type.' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		if ( ! current_user_can( $post_type->cap->edit_posts ) ) {
+			return new WP_Error(
+				'rest_cannot_read',
+				__( 'Sorry, you are not allowed to read post counts for this post type.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves post counts for a specific post type.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$post_type = $request['post_type'];
+		$counts    = wp_count_posts( $post_type );
+
+		if ( ! $counts ) {
+			return new WP_Error(
+				'rest_post_counts_error',
+				__( 'Could not retrieve post counts.' ),
+				array( 'status' => 500 )
+			);
+		}
+
+		$data = $this->prepare_item_for_response( $counts, $request );
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares post counts for response.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param object          $item    Post counts data.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data   = array();
+		$fields = $this->get_fields_for_response( $request );
+
+		foreach ( $fields as $field ) {
+			if ( property_exists( $item, $field ) ) {
+				$data[ $field ] = intval( $item->$field );
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filters post type counts data for the REST API.
+		 * Allows modification of the post type counts data right before it is returned.
+		 *
+		 * @since 6.8.0
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param object           $item     The original post counts object.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_post_counts', $response, $item, $request );
+	}
+
+	/**
+	 * Retrieves the post counts schema, conforming to JSON Schema.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
```

Comment:
I wonder what would happen if a new post status is added to the Core.
Should this list of supported types remain hardcoded?
If so, could a PHPUnit test method be added to trigger a failure if a new post status is missing from the schema? This would help ensure the schema stays updated.

Reply:
I think if there are custom stati, then they should be returned by this new endpoint. 

But more importantly, I just realized: this whole `get_item_schema()` is invalid since the endpoint in this PR only has a `get_items` endpoint for now. 🤦🏻 

I think I can safely remove and just inherit from `WP_Rest_Controller`